### ISSUE TITLE
Q: How to create an object by specifying a parent?

### ISSUE THREAD

**Issue by playdual** [BattlehubCode/RTE_Docs#133]
Hi.

As far as I know, when creating objects through RTÉditor, objects are created as children of objects called 'Scene'.

Instead of this "Scene" object, I want to create an object with the child of the object I want.

Ultimately, I press a button or toggle (for example, Toggle) 
In Hierarchy within RTÉditor, I would like to display only objects that fit that toggle in Scene and Hierarchy.

At the moment, when I deactivate a parent object, it seems that Hierarchy simply deactivates the child object together.

Current)
![image](https://github.com/user-attachments/assets/f21ca8ec-22d5-49a4-9e02-a84634174770)

Want)
![image](https://github.com/user-attachments/assets/dc9c5968-3a8d-4b56-aa53-d1e33094450e)

And when I turn on the toggle, the object that fits the toggle turns on! (so that it's visible)

To summarize, I need two pieces of information.

1) What object can I create by specifying a parent object when creating.

2) Can only certain objects be invisible/visible from the Hierarchy View and the Scene View within the RTÉditor.

Is this possible?

**Comment by BattlehubCode** [comment]
you can hide object in hierarchy by setting hide flags to HideInHierarchy
go.hideFlags = HideFlags.HideInHierarchy

or you can override HierarchyViewModel and display objects you need by overriding CanShowInHierarchy method for example.

```
using Battlehub.RTCommon;
using Battlehub.RTEditor.ViewModels;

namespace Battlehub.RTEditor.Examples
{
    public class ExtendHierarchyWindowExample : RuntimeWindowExtension
    {
        public override string WindowTypeName => BuiltInWindowNames.Hierarchy;

        protected override void Extend(RuntimeWindow window)
        {
            ViewModelBase.ReplaceWith<HierarchyViewModelOverride>(window);
        }
    }
}

using Battlehub.RTCommon;
using Battlehub.RTEditor.ViewModels;
using System.Collections.Generic;
using UnityEngine;
using UnityWeld.Binding;

namespace Battlehub.RTEditor.Examples
{
    [Binding]
    public class HierarchyViewModelOverride : HierarchyViewModel
    {
        protected override bool CanShowInHierarchy(ExposeToEditor obj)
        {
            if (obj.name.StartsWith("A"))
            {
                return false;
            }

            return base.CanShowInHierarchy(obj);
        }
```

The above should help you gather control over the hierarchy window. I probably misunderstood your first question, if so could you send me an example scene so I can better understand what you are doing with these toggles.
